### PR TITLE
Reduce the complexity of boolean_algebra/quine_mc_cluskey.py

### PR DESCRIPTION
**Pull request by MaximSmolskiy**
### Describe your change:

Reduce the complexity of boolean_algebra/quine_mc_cluskey.py from 17 to 13

* [ ] Add an algorithm?
* [x] Fix a bug or typo in an existing algorithm?
* [ ] Documentation change?

### Checklist:
* [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Python/blob/master/CONTRIBUTING.md).
* [x] This pull request is all my own work -- I have not plagiarized.
* [x] I know that pull requests will not be merged if they fail the automated tests.
* [x] This PR only changes one algorithm file.  To ease review, please open separate PRs for separate algorithms.
* [x] All new Python files are placed inside an existing directory.
* [x] All filenames are in all lowercase characters with no spaces or dashes.
* [x] All functions and variable names follow Python naming conventions.
* [x] All function parameters and return values are annotated with Python [type hints](https://docs.python.org/3/library/typing.html).
* [x] All functions have [doctests](https://docs.python.org/3/library/doctest.html) that pass the automated testing.
* [x] All new algorithms include at least one URL that points to Wikipedia or another similar explanation.
* [x] If this pull request resolves one or more open issues then the commit message contains `Fixes: #{$ISSUE_NO}`.

### CODE REVIEW

**Comment by MaximSmolskiy** [comment]
@cclauss please review

File: boolean_algebra/quine_mc_cluskey.py [hunk 1]
```diff
@@ -92,40 +92,34 @@ def selection(chart: list[list[int]], prime_implicants: list[str]) -> list[str]:
     temp = []
     select = [0] * len(chart)
     for i in range(len(chart[0])):
```
Comment (cclauss) [inline boolean_algebra/quine_mc_cluskey.py:94]: `range(len())` is almost always a sign that `enumerate()` should have been used instead.  Can you please review all uses of `range(len())` to see which of them can be converted?

Comment (MaximSmolskiy) [reply boolean_algebra/quine_mc_cluskey.py:94]: All code is based on indices and indices are used for assignment (and for accessing elements of several lists). So, for me `evaluate` doesn't help much here

File: boolean_algebra/quine_mc_cluskey.py [hunk 2]
```diff
@@ -92,40 +92,34 @@ def selection(chart: list[list[int]], prime_implicants: list[str]) -> list[str]:
...
-        count = 0
-        rem = -1
-        for j in range(len(chart)):
-            if chart[j][i] == 1:
-                count += 1
-                rem = j
+        count = sum(chart[j][i] == 1 for j in range(len(chart)))
```
Comment (cclauss) [inline boolean_algebra/quine_mc_cluskey.py:95]:
```suggestion
        count = sum(row[i] == 1 for row in chart))
```

File: boolean_algebra/quine_mc_cluskey.py [hunk 3]
```diff
@@ -92,40 +92,34 @@ def selection(chart: list[list[int]], prime_implicants: list[str]) -> list[str]:
...
         if count == 1:
+            rem = max(j for j in range(len(chart)) if chart[j][i] == 1)
```
Comment (cclauss) [inline boolean_algebra/quine_mc_cluskey.py:97]:
```suggestion
            rem = max(j for j, row in enumerate(chart) if row[i] == 1)
```

File: boolean_algebra/quine_mc_cluskey.py [hunk 4]
```diff
@@ -92,40 +92,34 @@ def selection(chart: list[list[int]], prime_implicants: list[str]) -> list[str]:
...
             select[rem] = 1
     for i in range(len(select)):
-        if select[i] == 1:
-            for j in range(len(chart[0])):
-                if chart[i][j] == 1:
-                    for k in range(len(chart)):
-                        chart[k][j] = 0
-            temp.append(prime_implicants[i])
+        if select[i] != 1:
```
Comment (cclauss) [inline boolean_algebra/quine_mc_cluskey.py:100]:
```suggestion
    for i, item in select:
        if item != 1:
```

File: boolean_algebra/quine_mc_cluskey.py [hunk 5]
```diff
@@ -92,40 +92,34 @@ def selection(chart: list[list[int]], prime_implicants: list[str]) -> list[str]:
...
+            continue
+        for j in range(len(chart[0])):
+            if chart[i][j] != 1:
```
Comment (cclauss) [inline boolean_algebra/quine_mc_cluskey.py:103]:
```suggestion
            if item[j] != 1:
```

File: boolean_algebra/quine_mc_cluskey.py [hunk 6]
```diff
@@ -92,40 +92,34 @@ def selection(chart: list[list[int]], prime_implicants: list[str]) -> list[str]:
...
+                continue
+            for k in range(len(chart)):
+                chart[k][j] = 0
```
Comment (cclauss) [inline boolean_algebra/quine_mc_cluskey.py:106]:
```suggestion
            for row in chart:
                row[j] = 0
```

File: boolean_algebra/quine_mc_cluskey.py (hunk 1 above)
Comment (cclauss) [reply boolean_algebra/quine_mc_cluskey.py:94]: hogwash.

**Review by cclauss — DISMISSED** [review]

**Review by tianyizheng02 — CHANGES_REQUESTED** [review]
Your changes look good, and I only have a couple of minor suggestions. However, I noticed a number of issues elsewhere in the code (not related to your changes) that require refactoring and renaming some things. I don't want to hold this PR up, so it's probably best that I address those issues in a later PR.

**Review by tianyizheng02 — APPROVED** [review]

**Pull request merged**